src/components/engagement/engagement.service.ts

```diff
@@ -189,6 +189,48 @@ export class EngagementService {
     return results?.type as ProjectType | undefined;
   }
 
+  protected async getIEByProjectAndIntern(
+    projectId: string,
+    internId: string
+  ): Promise<InternshipEngagement | undefined> {
+    const result = await this.db
+      .query()
+      .match([node('intern', 'User', { active: true, id: internId })])
+      .match([node('project', 'Project', { active: true, id: projectId })])
+      .match([
+        node('project'),
+        relation('out', '', 'engagement'),
+        node('internshipEngagement'),
+        relation('out', '', 'intern'),
+        node('intern'),
+      ])
+      .return('internshipEngagement.id as id')
+      .first();
+
+    return (result as unknown) as InternshipEngagement | undefined;
```

Comment:
Since you are just doing a boolean check on this result you mind as well convert to boolean here instead of faking TS type.